src/libasr/runtime/lfortran_intrinsics.c

```diff
@@ -1165,3 +1165,24 @@ LFORTRAN_API void _lpython_close(int64_t fd)
         exit(1);
     }
 }
+
+// Command line arguments
+int32_t argc;
+char **argv;
+
+LFORTRAN_API void _lpython_set_argv(int32_t argc_1, char **argv_1) {
+    argv = malloc(argc_1 * sizeof(char *));
+    for (size_t i = 0; i < argc_1; i++) {
+        argv[i] = strdup(argv_1[i]);
+    }
+    argc = argc_1;
+}
+__attribute__((section(".init_array"))) static void *ctr = &_lpython_set_argv;
```

Comment:
This doesn't seem to work in macOS and Windows.

Reply:
Just use regular C, not these `__attribute__` or stuff.

I think we either have to call some `libc` functions to access the arguments, or we need to emit this `_lpython_set_argv` function in the ASR->LLVM/C backends when the program starts.

Reply:
Looks like there is no easy way to do that: https://stackoverflow.com/questions/2471553/access-command-line-arguments-without-using-char-argv-in-main#24718544

So I recommend we simply get those arguments in `main` (both in C and in LLVM), and then call our function like `_lcompilers_set_argv`, to set these in our "internal structure", like you did above. That seems like a portable, clean approach that should work everywhere.